include/internal/refcount.h

```diff
@@ -134,14 +134,14 @@ static __inline int CRYPTO_DOWN_REF(volatile int *val, int *ret,
 static __inline int CRYPTO_UP_REF(volatile int *val, int *ret,
                                   ossl_unused void *lock)
 {
-    *ret = _InterlockedExchangeAdd(val, 1) + 1;
+    *ret = _InterlockedExchangeAdd((long volatile *)val, 1) + 1;
```

Comment:
If `long` is wider than `int`, this has the potential to modify the word after `val`.

Reply:
This is a Windows-specific API - I'm now aware of any (current) Windows platforms where `sizeof(long)` != `sizeof(int)`

Reply:
Ok, sorry for the distraction. It's tough keeping track; on most other platforms `long` is 64 bits while `int` is only 32.